### ISSUE TITLE
Add components to centos CI containers for reproducibility

### ISSUE THREAD

**Issue by williamfgc** [QMCPACK/qmcpack#4512]
**Is your feature request related to a problem? Please describe.**
In order to debug CI failures, e.g. #4510 the container images should be interactive and reproducible.
After downloading and running the container it's currently missing the spack installation used to installed required packages.

**Describe the solution you'd like**
The location `/opt/spack` from [here](https://github.com/QMCPACK/qmcpack/blob/develop/tests/test_automation/github-actions/ci/run_step.sh#L109) is missing. Also `find /opt -name "spack"` returns empty.

**Describe alternatives you've considered**
N/A 

**Additional context**
Not sure if this is intentional when using this spack [image](https://github.com/QMCPACK/qmcpack/blob/develop/config/docker/dependencies/centos-stream/Dockerfile#L2).

**Comment by williamfgc** [comment]
@walshmm I might need your guidance on this. Is spack supposed to be available in the image? `/opt/spack` is added in the PATH and LD_LIBRARY_PATH when running CI (see first message), so I thought it would be available if I pull the container locally. Does it make sense?

**Comment by williamfgc** [comment]
OK, thanks for clarifying. Should non-existing directories remove from PATH and LD_LIBRARY_PATH?

**Comment by walshmm** [comment]
Yes, for clarity's sake.  Leftovers from trying to figure out the right env for the runners.

**Comment by williamfgc** [comment]
@walshmm thanks, feel free to do a PR. The reason we want to keep `run_step.sh` clean is so we can reproduce locally potential CI failures.

**Comment by williamfgc** [comment]
@prckent good point, @walshmm please add the spack centos image to this [page](https://qmcpack.readthedocs.io/en/develop/running_docker.html).
